src/main/java/org/wso2/okta/client/OktaOAuthClient.java

```diff
@@ -0,0 +1,1046 @@
+            APIManagementException {
+        CloseableHttpClient httpClient = HttpClientBuilder.create().build();
+        BufferedReader reader = null;
+
+        try {
+            String oktaInstanceUrl = configuration.getParameter(OktaConstants.OKTA_INSTANCE_URL);
+            String authorizationServerId = configuration.getParameter(OktaConstants.OKTA_AUTHORIZATION_SERVER_ID);
+            HttpPost httpPost = new HttpPost(oktaInstanceUrl + OktaConstants.OAUTH2 + authorizationServerId +
+                    OktaConstants.TOKEN_ENDPOINT);
+            httpPost.setEntity(new UrlEncodedFormEntity(parameters));
+            String encodedCredentials = getEncodedCredentials(clientId, clientSecret);
+
+            httpPost.setHeader(OktaConstants.AUTHORIZATION, OktaConstants.AUTHENTICATION_BASIC + encodedCredentials);
+            if (log.isDebugEnabled()) {
+                log.debug("Invoking HTTP request to get the accesstoken.");
+            }
+            HttpResponse response = httpClient.execute(httpPost);
+            int statusCode = response.getStatusLine().getStatusCode();
+            HttpEntity entity = response.getEntity();
+            reader = new BufferedReader(new InputStreamReader(entity.getContent(), OktaConstants.UTF_8));
+            JSONObject responseJSON = getParsedObjectByReader(reader);
+
+            if (HttpStatus.SC_OK == statusCode) {
+                if (responseJSON != null) {
+                    if (log.isDebugEnabled()) {
+                        log.debug(String.format("JSON response after getting new access token: %s",
+                                responseJSON.toJSONString()));
+                    }
+                    return responseJSON;
+                }
+            } else {
+                log.error(String.format("Failed to get accessToken for Consumer Key %s. Response: %s", clientId,
+                        responseJSON.toJSONString()));
+            }
+        } catch (UnsupportedEncodingException e) {
+            handleException(OktaConstants.ERROR_ENCODING_METHOD_NOT_SUPPORTED + e.getMessage(), e);
+        } catch (ParseException e) {
+            handleException(OktaConstants.ERROR_WHILE_PARSE_RESPONSE + e.getMessage(), e);
+        } catch (IOException e) {
+            handleException(OktaConstants.ERROR_OCCURRED_WHILE_READ_OR_CLOSE_BUFFER_READER + e.getMessage(), e);
+        } finally {
+            closeResources(reader, httpClient);
+        }
+        return null;
+    }
+
+    /**
+     * Update the access token info after getting new access token.
+     *
+     * @param tokenInfo    Token info need to be updated.
+     * @param responseJSON AccessTokenInfo
+     * @return AccessTokenInfo
+     */
+    private AccessTokenInfo updateTokenInfo(AccessTokenInfo tokenInfo, JSONObject responseJSON) {
+        if (log.isDebugEnabled()) {
+            log.debug("Update the access token info after getting new access token.");
```

Comment:
This debug log does not give anything meaningful. When adding logs, we should include something specific to that call ? May be responseJSON or tokenInfo